syntax = "proto3";
package store;

message GetObjectRequest {
    string object_ref = 1; // reference to the object to retrieve
}

message GetObjectResponse {
    bytes object_data = 1; // serialized object data
    string error = 2; // error message, if any
}

message PutObjectRequest {
    bytes object_data = 1; // serialized object data to store
}

message PutObjectResponse {
    string object_ref = 1; // reference to the stored object
    string error = 2; // error message, if any
}

service StoreService {
    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {}
    rpc PutObject(PutObjectRequest) returns (PutObjectResponse) {}
}